### PR TITLE
Add build mode e2e test to CircleCI

### PR DESCRIPTION
**Pull request by gytis**

### CODE REVIEW

**Comment by gytis** [comment]
Thanks @metacosm . docker-image build is working again

**Comment by metacosm** [comment]
@gytis good to review?

**Comment by gytis** [comment]
@metacosm We've been having various issues, thus so many commits and such a delay. The latest one is with DNS config. I'll ping you for a review once I have all the variations working.

**Comment by gytis** [comment]
So far I go OpenShift working. Kubernetes still has docker registry issue so I propose to review this PR and merge it in once accepted. Then we can create a new one once we can have Kubernetes working

File: .circleci/config.yml [hunk 1]
```diff
@@ -155,13 +144,20 @@ commands:
...
-            echo "Running e2e test in a '<< parameters.namespace >>' namespace of '<< parameters.cluster-ip >>' in a '<< parameters.mode >>' mode"
+            echo "Running '<< parameters.mode >>' mode e2e test in a '<< parameters.namespace >>' namespace"
             cd demo
-            ./mvnw package
-            ./scripts/end-to-end_fruit_demo.sh << parameters.cluster-ip >> << parameters.namespace >> << parameters.mode >>
+
+            ./mvnw clean package
+
+            # Set a correct deploymentMode in a component configuration
+            sed -i -E "s/(deploymentMode:).*/\1 \"<< parameters.mode >>\"/g" fruit-backend-sb/target/classes/META-INF/dekorate/component.yml
```
Comment (cmoulliard) [inline .circleci/config.yml]: if you use the new API -> halkyon.io and snapshot build of dekorate, then the file name to be used is now `halkyon.yml` and not anymore `component.yml`

**Comment by cmoulliard** [comment]
What do you do to switch from `dev` to `build` mode for the end to end test bash script as demo project will by default generate a `halkyon.yml` file containing as `deploymentMode: dev` ? 
Bash script should not include anymore the steps to do `oc adm policy ...` as this is now managed by the operator ?
@gytis

File: .circleci/config.yml [hunk 2]
```diff
@@ -155,13 +144,20 @@ commands:
...
-            echo "Running e2e test in a '<< parameters.namespace >>' namespace of '<< parameters.cluster-ip >>' in a '<< parameters.mode >>' mode"
+            echo "Running '<< parameters.mode >>' mode e2e test in a '<< parameters.namespace >>' namespace"
             cd demo
-            ./mvnw package
-            ./scripts/end-to-end_fruit_demo.sh << parameters.cluster-ip >> << parameters.namespace >> << parameters.mode >>
+
+            ./mvnw clean package
+
+            # Set a correct deploymentMode in a component configuration
+            sed -i -E "s/(deploymentMode:).*/\1 \"<< parameters.mode >>\"/g" fruit-backend-sb/target/classes/META-INF/dekorate/halkyon.yml
```
[the remark this answers is not in the document]
Comment (gytis) [reply .circleci/config.yml]: `<< parameters.mode >>` is a special CircleCI syntax to access command arguments. Locally you need to replace it with an actual value manually or at least with an env variable.

**Comment by cmoulliard** [comment]
LGTM. It can be merged now @gytis

**Pull request merged**